frb_codegen/src/main.rs

```diff
@@ -14,20 +12,21 @@ fn main() -> anyhow::Result<()> {
     #[cfg(feature = "serde")]
     let dump_config = raw_opts.dump.clone();
 
-    let configs = config_parse(raw_opts);
-    debug!("configs={:?}", configs);
+    let (all_configs, all_symbols) = config_parse(raw_opts)?;
+    debug!("configs={:?}", all_configs);
 
+    // generation of rust api for ffi
     #[cfg(feature = "serde")]
     if let Some(dump) = dump_config {
-        return dump::dump_multi(&configs, dump).context("Failed to dump config");
+        return dump::dump_multi(&all_configs, dump).context("Failed to dump config");
     }
 
-    // generation of rust api for ffi
-    let all_symbols = get_symbols_if_no_duplicates(&configs)?;
-
+    // If in multi-blocks case, the shared block MUST be generated at first.
+    // Otherwise, the generated dart code for regular blocks would be problematic, since when
+    // the shared rust module has not yet been generated.
```

Comment:
```suggestion
    // If in multi-blocks case, the shared block MUST be generated at first.
    // Otherwise, the generated dart code for regular blocks would be problematic, since when they are being generated,
    // the shared rust module they depend on has not yet been generated.
```